doc/jacoco.md

```diff
@@ -0,0 +1,164 @@
+# Importing JaCoCo coverage report in XML format
+
+Version 5.12 of our SonarJava analyzer deprecated use JaCoCo's binary format (`.exec` files) to import coverage. This binary format is internal to the JaCoCo project, and as such there are no guarantees for backward compatibility, so it should not be used for integration purposes. 
+
+As a replacment, we developed the [sonar-jacoco](https://docs.sonarqube.org/display/PLUG/JaCoCo+Plugin) plugin, which imports JaCoCo's XML coverage report, and this is the preferred option now. In this guide, I will describe how to import this XML report in some common scenarios.
+
+You can find sample projects using the setup described here in [this repository](https://github.com/SonarSource/sonar-scanning-examples).
+
+## Maven
+
+Remove any use of the `sonar.jacoco.itReportPath`, `sonar.jacoco.reportPath`, and `sonar.jacoco.reportMissing.force.zero` properties; they are deprecated and related functionality will be removed in the future.
+
+We will use the [jacoco-maven-plugin](https://www.jacoco.org/jacoco/trunk/doc/maven.html) and its [report goal](https://www.eclemma.org/jacoco/trunk/doc/report-mojo.html) to create a code coverage report. Usually, you would want to create a specific profile which executes unit tests with the JaCoCo agent and creates a coverage report. This profile would then only be activated if coverage information is requested (e.g., in the CI pipeline).
+
+In the most basic case we will need to execute two goals: `jacoco:prepare-agent` which creates the command line argument to be used for JVM during unit test execution, and `jacoco:report` which uses data collected during unit test execution to generate a report in html, xml or csv format.
```

Comment:
I don't understand what the "for JVM" part of this means:
"which creates the command line argument to be used for JVM"

Reply:
changed to "which creates the command line argument for JVM running the tests "